ports/openldap/openssl.patch

```diff
@@ -1,13 +1,35 @@
 diff --git a/configure.ac b/configure.ac
-index 626d024..c1eb141 100644
+index 626d024..e97f548 100644
 --- a/configure.ac
 +++ b/configure.ac
-@@ -1235,7 +1235,7 @@ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
+@@ -1223,6 +1223,9 @@ fi
  
- 		AC_CHECK_LIB(ssl, SSL_export_keying_material_early,
- 			[have_openssl=yes], [have_openssl=no],
--			[-lcrypto])
-+			[-lcrypto -lpthread])
+ ol_link_tls=no
+ if test $ol_with_tls = openssl || test $ol_with_tls = auto ; then
++	PKG_CHECK_MODULES(OPENSSL, openssl >= 1.1.1, [have_openssl=yes], [AC_MSG_FAILURE([OpenSSL is required])])
++	CFLAGS="${OPENSSL_CFLAGS} ${CFLAGS}"
++	ac_cv_lib_ssl_SSL_export_keying_material_early=yes
```

Comment:
openldap has good checking for ssl / tsl during the config step already (version checking, TLS v SSL support, missing SSL etc.).  IMO just fixing the AC_CHECK above to work with the static lib is sufficient.

Reply:
No, it is not at good enough at all *for static linkage*. Because it would mean to know the transitive usage requirements in advance. That's why `pkg-config` is so useful, when it is properly exported.